SRVKS-163: OpenShift ClusterLogging with Knative (#16224)
openshift/openshift-docs: modules/srvrlss-using-cluster-logging-find-knative-serving-logs.adoc

// Module included in the following assemblies:
//

> // cluster-logging-serverless.adoc

Reviewer: "serverless/cluster-logging-serverless.adoc" - the assembly should be in the 'serverless' folder.

[id=using-cluster-logging-to-find-Knative-Serving-logs_{context}"]
= Using Cluster Logging to find logs for Knative Serving components

.Prerequisites

> * xref:../efk-logging-about_{context}[Cluster Logging components] are installed.

Reviewer 2: I would suggest adding the link to appropriate assembly the additional resources section of the assembly this module will build in since we can't link module to module.

Reviewer: I'd probably just include the re-used module to save confusion? (example in https://github.com/openshift/openshift-docs/pull/16154/files#diff-9ef19605df4e8cbc8ccf698f7e10aadeR13 if you're interested)

.Procedure

> . To open the Kibana UI (the visualization tool for Elasticsearch), use the following command to get the Kibana route:

Reviewer 2: I would suggest removing the parentheses.

Reviewer: Also idk if we're supposed to use `:` at the end of sentences?

> `oc -n openshift-logging get route kibana`

Reviewer 2: The command should go in a code block preceded by a "+" (the "+" will keep the code block in line with the numbered list). 
+
`-` x 4 (4 hyphens `----` - Github does some wonky formatting)
$ oc -n openshift-logging get route kibana
`-` x 4

. Use the route's URL to navigate to the Kibana dashboard and log in.

. Ensure the index is set to *.all*. If the index is not set to *.all*, only the OpenShift system logs will be listed. 

> . You can filter the logs by using the knative-serving namespace. Enter `kubernetes.namespace_name:knative-serving` in the search box to filter results.

Reviewer 2: `knative-serving`

Reviewer: I'd actually go with 'Knative Serving namespace' unless we're specifying some config or something?

+

> NOTE: Knative Serving uses structured logging by default. You can enable the parsing of these logs by customizing the xref:../efk-logging-fluentd-json_{context}[ClusterLogging fluentd settings]. This makes the logs more searchable and enables filtering on the log level to quickly identify issues.

Reviewer 2: [NOTE]
`=` x 4 (same issue as with the code block `====`)
Knative Serving uses structured logging by default. You can enable the parsing of these logs by customizing the ClusterLogging fluentd settings. This makes the logs more searchable and enables filtering on the log level to quickly identify issues.
`=` x 4

I would suggest adding the link to appropriate assembly the additional resources section of the assembly this module will build in since we can't link module to module.

// Module included in the following assemblies:
//
// cluster-logging-serverless.adoc

[id=using-cluster-logging-to-find-Knative-Serving-logs_{context}"]

> = Using Cluster Logging to find Knative Serving logs

Reviewer 3: ```suggestion
= Using Cluster Logging to find logs for Knative Serving components
```

> Knative Serving provides default logs for your application. 

Reviewer 3: This talks about the knative serving components specifically and has nothing to do with applications.

.Prerequisites

* xref:../efk-logging-about_{context}[Cluster Logging components] are installed.

> .Procedure

Reviewer 3: Most of this procedure simply covers "how to get to kibana" and should be there only once I think? Below's file has an exact duplicate.

> . To open the Kibana UI (the visualization tool for Elasticsearch), you must start a local proxy by running the following command to get the Kibana route:

Reviewer 3: The following command does not start a local proxy.

`oc -n openshift-logging get route kibana`

> . Log into the Kibana dashboard using the Kibana route as the URL.

Reviewer 3: ```suggestion
. Use the route's URL to navigate to the Kibana dashboard and log in.
```
